x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CachedBlobContainerIndexInput.java

```diff
@@ -136,53 +146,221 @@ protected void readInternal(ByteBuffer b) throws IOException {
         final long position = getFilePointer() + this.offset;
         final int length = b.remaining();
 
-        int totalBytesRead = 0;
-        while (totalBytesRead < length) {
```

Comment:
In fact we (almost) never looped here, the inner read methods always satisfied the complete read. The only case where we did loop is if the read spanned two 32MB-aligned ranges, but in that case we may as well read both ranges in one go anyway so that's what we do now.